### PR TITLE
add tf workflow

### LINKED ISSUE #1: dummy requirement 1

### LINKED ISSUE #2: dummy requirement 2
Describe the requirement with an explicit title and if necessary add details in the body here.

You can add a label "requirement-topic:my-topic" which can help to organize requirements within your project.

### PR DESCRIPTION
**Pull request by sjoshi-jpl**
## 🗒️ Summary
Added terraform cicd workflow

## ⚙️ Test Data and/or Report
One of the following should be included here:
* Reference to regression test included in code (preferred wherever reasonable)
* Attach test data here + outputs of tests

## ♻️ Related Issues

Ref - https://github.com/NASA-PDS/devops/issues/52

### CODE REVIEW

**Comment by sjoshi-jpl** [comment]
Note: the tests will fail since it's looking for repository variables which cannot be supplied as these are just example templates.

File: .github/workflows/terraform-cicd.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+name: 'Terraform-Workflow'
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    types: [opened, edited, reopened, ready_for_review, review_requested]
```
Comment (jordanpadams) [inline .github/workflows/terraform-cicd.yaml:8]: @sjoshi-jpl do we want to run these TF workflows on PRs/branches? I was thinking we would only want to deploy these things when they are merged to main?

**Review by jordanpadams — CHANGES_REQUESTED** [review]

File: .github/workflows/terraform-cicd.yaml (hunk 1 above)
Comment (sjoshi-jpl) [reply .github/workflows/terraform-cicd.yaml:8]: @jordanpadams yes but I can limit the number of times this workflow gets executed by only running `terraform plan` on PR creation and push to main.

[more details](https://github.com/NASA-PDS/nucleus/pull/56#pullrequestreview-1522517881)

**Review by nutjob4life — APPROVED** [review]

**Comment by jordanpadams** [comment]
@sjoshi-jpl I added `workflow_dispatch:` to this and the other 2 PRs. if you agree with that, go ahead with a merge.

**Pull request merged**